### PR TITLE
Cleanup how item logistics handles null/empty values

### PR DESCRIPTION
**Pull request by ziadoz**
The new `setItemLogistics()` function should not require any parameters (at least in the short term). I've cleaned up the internals so the corresponding elements only get created if a value was passed. I also corrected msFamId to msfamId, which matches the Pangaea element name, as it's a little confusing to have it cased differently.

### CODE REVIEW

File: src/Item.php [hunk 1]
```diff
@@ -253,28 +253,35 @@ public function setAssets(array $urls, $urlPrefix = '')
...
         ];
 
+        $itemLogisticsElements   = [];
         $itemLogisticsAttributes = [];
 
-        foreach ($itemLogistics as $key => $value) {
+        foreach ($itemLogisticsParams as $key => $value) {
+            $itemLogisticsElements[$key] = '';
+
+            if (! empty($value)) {
```
Comment (gabriel403) [inline src/Item.php]: We're trying (albeit slowly) to get away from using empty, check explicitly for what you want, is_null/strlen etc

File: src/Item.php [hunk 2]
```diff
@@ -340,7 +343,7 @@ public function setItemLogistics($legacyDistributorId, $mdsFamId, $vendorStockId
 <!-- END: Required Dummy Values -->
 <shipNodes>
     <shipNode>
-        <legacyDistributorId>{$itemLogistics['legacyDistributorId']}</legacyDistributorId>
+        {$itemLogisticsElements['legacyDistributorId']}
```
Comment (gabriel403) [inline src/Item.php]: This will mean sometimes there are blank lines in the xml do these get removed later or is it ok to have them?

**Comment by gabriel403** [comment]
Other than comments :+1:

**Comment by gabriel403** [comment]
@ziadoz I added semaphore integration to this now, you can see it failed, suspect it's due to the output folder not being present, we either need to create it in the tests or commit an empty dir with just .gitkeep file in it.

File: src/Item.php (hunk 2 above)
Comment (ziadoz) [inline src/Item.php]: The output gets run through `DOMDocument` with settings to format the output, so it won't appear once rendered.

File: src/Item.php (hunk 1 above)
Comment (ziadoz) [inline src/Item.php]: I've replaced all usage of `empty()` with `mb_strlen()` instead.

**Comment by gabriel403** [comment]
:+1: should be able to get away with a x.x.1 release

**Comment by ziadoz** [comment]
I've made one more minor tweak to this to stop empty values being added to the attributes group.
